build.gradle

```diff
@@ -77,6 +78,19 @@ allprojects {
   group = 'org.opensearch'
   version = VersionProperties.getOpenSearch()
   description = "OpenSearch subproject ${project.path}"
+
+  afterEvaluate {
+    project.tasks.withType(Test) { task ->
+      // This is so hacky: now, by default, test tasks uses JUnit framework and always includes 'junit'
+      // JARs from the Gradle distribution (no ways to override this behavior). It causes JAR hell on test 
+      // classpath, example of the report:
+      //
+      //  jar1: /home/ubuntu/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar
+      //  jar2: /home/ubuntu/.gradle/wrapper/dists/gradle-8.0-rc-1-all/2p8rgxxewg8l61n1p3vrzr9s8/gradle-8.0-rc-1/lib/junit-4.13.2.jar
+      // 
+      task.getTestFrameworkProperty().convention(getProviderFactory().provider(() -> new JUnitTestFramework(task, task.getFilter(), false)));
```

Comment:
I understand the problem here (gradle includes JUnit by default resulting in a duplicate JAR on the classpath). But how does code solve it? I obviously don't know gradle very well...

Reply:
> But how does code solve it? 

You see the last option in the `JUnitTestFramework` constructor - `false` - it literally says "do not include the JUnit dependencies" [1] 

[1] https://github.com/gradle/gradle/blob/master/subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestFramework.java#L44

Reply:
Makes sense. Thanks! One of Nick's cryptic booleans :)

Reply:
Yeah, it was possible to control this behaviour with a property - but not anymore with Gradle 8